flake8_trio.py

```diff
@@ -36,66 +36,206 @@ def make_error(error: str, lineno: int, col: int, *args: Any, **kwargs: Any) ->
     return (lineno, col, error.format(*args, **kwargs), type(Plugin))
 
 
-def is_trio_call(node: ast.AST, *names: str) -> Optional[str]:
+class TrioScope:
+    def __init__(self, node: ast.Call, funcname: str, packagename: str):
+        self.node = node
+        self.funcname = funcname
+        self.packagename = packagename
+        self.variable_name: Optional[str] = None
+        self.shielded: bool = False
+        self.timeout: bool = False
```

Comment:
Let's name this `has_timeout`, to avoid confusing it for the length of a timeout.